compiler/modules/front.syntax/tests/testParserOnTestSuite.bal

```diff
@@ -22,6 +22,25 @@ function testParserOnTestSuite() returns err:Syntax|io:Error|file:Error? {
                 continue;
             }
             else {
+                [err:Position, err:Position][] topLevelDefnPos = [];
+                foreach ImportDecl decl in part.importDecls {
+                    topLevelDefnPos.push([decl.startPos, decl.endPos]);
+                }
+                foreach ModuleLevelDefn defn in part.defns {
+                    topLevelDefnPos.push([defn.startPos, defn.endPos]);
+                }
+                topLevelDefnPos = topLevelDefnPos.sort();
+                err:Position lastEnd = 1<<32;
+                foreach var [startPos, endPos] in topLevelDefnPos {
+                    test:assertTrue(startPos >= lastEnd, "overlapping top level definitions");
+                    test:assertTrue(startPos <= endPos, "invalid start and end positions");
+                    SourceFile file = part.file;
+                    var [stLine, stCol] = file.lineColumn(startPos);
+                    var [endLine, endCol] = file.lineColumn(lastEnd);
+                    string errorBody = string ` filename: ${file.filename()} between (${endLine}, ${endCol})  and (${stLine}, ${stCol})`;
+                    test:assertTrue(testIsWhitespace(part.file, lastEnd, startPos), "none white space tokens between top level definition" + errorBody);
```

Comment:
You don't have to have whitespace between top-level definitions:

const X = 1;const Y = 2;

The point is that if there is any between, it should not be between the startPos/endPos of any of the definitions.

Reply:
I  am considering the the range from the last top level definition's endposition (or begining of the file for the first definition) and start position of current definition.  In a case like this where this range is empty `testIsWhitespace` will return true.